src/Sign.Core/Native/mansign2.cs

```diff
@@ -386,28 +364,14 @@ static byte[] ComputeHashFromManifest(XmlDocument manifestDom, bool oldFormat, b
                 var exc = new XmlDsigExcC14NTransform();
                 exc.LoadInput(normalizedDom);
 
-                if (useSha256)
+                using var sha2 = SHA256.Create();
+                var hash = sha2.ComputeHash(exc.GetOutput() as MemoryStream);
+                if (hash == null)
                 {
-                    using var sha2 = SHA256.Create();
-                    var hash = sha2.ComputeHash(exc.GetOutput() as MemoryStream);
-                    if (hash == null)
-                    {
-                        throw new CryptographicException(Win32.TRUST_E_BAD_DIGEST);
-                    }
-
-                    return hash;
+                    throw new CryptographicException(Win32.TRUST_E_BAD_DIGEST);
```

Comment:
Can we extract this logic into a method so that it can be called here and line no:367 to 371?

Reply:
Good feedback, but I would like to hold off on that change.

This file and mansign.cs were once copies of .NET Framework source code.  See https://github.com/microsoft/referencesource/blob/master/inc/mansign.cs and https://github.com/microsoft/referencesource/blob/master/inc/mansign2.cs.

I've tried to maintain a high bar for making changes to these files.  I've added copyright headers, suppressed some warnings (that would be errors with treat warnings as errors enabled), and deleted code.  I've not generally refactored or improved the code, because long-term I think we should get rid of it and migrate to an actively maintained library that does manifest signing.  See https://github.com/dotnet/sign/issues/479.

Reply:
If you feel strongly about this, let's talk.